libsearch/secure_index_builder.go

```diff
@@ -76,12 +77,24 @@ func (sib *SecureIndexBuilder) buildBloomFilter(nonce uint64, document *os.File)
 // Blinds the bloom filter by setting random bits to be on for `numIterations`
 // iterations.
 func (sib *SecureIndexBuilder) blindBloomFilter(bf bitarray.BitArray, numIterations int64) error {
-	for i := int64(0); i < numIterations; i++ {
-		randUint, err := RandUint64n(sib.size)
+	i := numIterations
+	mask := BuildMaskWithLeadingZeroes(GetNumLeadingZeroes(sib.size))
+	for i > 0 {
+		randNums := make([]uint64, int64(float64(i)*1.3))
+		err := binary.Read(rand.Reader, binary.LittleEndian, &randNums)
```

Comment:
Please document the magic 1.3 number (and possibly move it out into its own constant).